flink-runtime/src/test/java/org/apache/flink/runtime/blob/BlobCacheSizeTrackerTest.java

```diff
@@ -0,0 +1,286 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License
+ */
+
+package org.apache.flink.runtime.blob;
+
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.configuration.BlobServerOptions;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.util.TestLogger;
+import org.apache.flink.util.concurrent.FutureUtils;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import javax.annotation.Nullable;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static org.apache.flink.runtime.blob.BlobServerPutTest.put;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+/** Tests for {@link BlobCacheSizeTracker}. */
+public class BlobCacheSizeTrackerTest extends TestLogger {
+
+    private static final int blobSize = 10_000;
+
+    private static final Random random = new Random();
+
+    private static final TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    @BeforeClass
+    public static void setUp() throws IOException {
+        temporaryFolder.create();
+    }
+
+    @AfterClass
+    public static void cleanUp() throws IOException {
+        temporaryFolder.delete();
+    }
+
+    @Test
+    public void testSizeLimitTrackerStatusUpdate() {
+        BlobCacheSizeTracker tracker = new BlobCacheSizeTracker(5L);
+        List<JobID> jobIds = new ArrayList<>();
+        List<BlobKey> blobKeys = new ArrayList<>();
+        for (int i = 0; i < 5; i++) {
+            jobIds.add(new JobID());
+            blobKeys.add(BlobKey.createKey(BlobKey.BlobType.PERMANENT_BLOB));
+        }
+        for (int i = 0; i < 5; i++) {
+            tracker.track(jobIds.get(i), blobKeys.get(i), 1);
+        }
+        tracker.update(jobIds.get(1), blobKeys.get(1));
+        tracker.update(jobIds.get(2), blobKeys.get(2));
+
+        List<Tuple2<JobID, BlobKey>> blobsToDelete = tracker.checkLimit(2);
+
+        assertThat(
+                blobsToDelete,
+                containsInAnyOrder(
+                        Tuple2.of(jobIds.get(0), blobKeys.get(0)),
+                        Tuple2.of(jobIds.get(3), blobKeys.get(3))));
+    }
+
+    @Test
+    public void testTrackPermanentBlobCache() throws Exception {
```

Comment:
It is test of `PermanentBlobCache` and should not be in `BlobCacheSizeTrackerTest`.